dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java

```diff
@@ -994,6 +1000,311 @@ public void selfRelatedContents() throws DotDataException {
+
+            APILocator.getContentletAPI().checkin(contentlet_1, APILocator.systemUser(), false);
+
+            try {
+                APILocator.getContentletAPI().checkin(contentlet_2, APILocator.systemUser(), false);
+                throw new AssertionError("DotRuntimeException Expected");
+            } catch (DotRuntimeException e) {
+                assertEquals(e.getMessage(),
+                        "Contentlet with id:`Unknown/New` and title:`` has invalid / missing field(s).");
+            }
+        } finally {
+            Config.setProperty("unique_per_site", uniquePerSite);
+        }
+    }
+
+
+    /**
+     * Method to test: {@link ContentletAPI#checkin(Contentlet, User, boolean)}  }
+     * When:
+     * - Create a {@link ContentType} with a unique field
+     * - Create two  {@link Contentlet} with the same value in the unique field in different hosts
+     *
+     * Should: Save successfully the two {@link Contentlet}
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @Test
+    public void savingFieldWithUniqueFieldInDifferentHost() throws DotDataException, DotSecurityException {
+        final Field uniqueTextField = new FieldDataGen()
+                .unique(true)
+                .type(TextField.class)
+                .next();
+
+        final ContentType contentType = new ContentTypeDataGen()
+                .field(uniqueTextField)
+                .nextPersisted();
+
+        final Host host1 = new SiteDataGen().nextPersisted();
+        final Host host2 = new SiteDataGen().nextPersisted();
+
+        final Contentlet contentlet_1 = new ContentletDataGen(contentType)
+                .host(host1)
+                .setProperty(uniqueTextField.variable(), "unique-value")
+                .next();
+
+        final Contentlet contentlet_2 = new ContentletDataGen(contentType)
+                .host(host2)
+                .setProperty(uniqueTextField.variable(), "unique-value")
+                .next();
+
+        APILocator.getContentletAPI().checkin(contentlet_1, APILocator.systemUser(), false);
+        APILocator.getContentletAPI().checkin(contentlet_2, APILocator.systemUser(), false);
+
+        final Optional<Contentlet> contentlet1FromDB = APILocator.getContentletAPI()
+                .findInDb(contentlet_1.getInode());
+
+        assertTrue(contentlet1FromDB.isPresent());
+
+        final Optional<Contentlet> contentlet2FromDB = APILocator.getContentletAPI()
+                .findInDb(contentlet_2.getInode());
+
+        assertTrue(contentlet2FromDB.isPresent());
+
+    }
+
+    /**
+     * Method to test: {@link ContentletAPI#checkin(Contentlet, User, boolean)}  }
+     * When:
+     * - Create a {@link ContentType} with a unique field
+     * - Create two  {@link Contentlet} with the same value in the unique field in different hosts
+     * - set the unique_per_site properties to false
+     *
+     * Should: Save successfully the two {@link Contentlet}
```

Comment:
bad `Should` - it should be error expected

Reply:
done https://github.com/dotCMS/core/pull/21843/commits/e20c14f55318f719c36d72a7c953bf8d9ac42dfb#diff-093de07a53a725b587963fd60c024f52e089c4f78e7ea88ba8e3a3a0cdaecab9R1156